### PR TITLE
fix: render only the currently open detached widgets.

### PR DESCRIPTION
**Pull request by marks0351**
[workerB](https://workerb.linearb.io/v2/badge/collaboration-page?magicLinkId=M62wKah)
## Description
Currently we render all the detached widgets(Modal) via a hook and the widget shows up when the metaValue for visibility is toggled for a modal.

Problems with this:
- we render all modal widgets even before they are needed.
- everytime something on the canvas changes the widgets are rendered.

To solve this we are going to rely on a state that captures the currently open detached widgets in anvil `anvilDetachedWidgets.currentlyOpenDetachedWidgets`
This is captured and reset currently based on show and close operations of a modal.

we now have the modals that are currently open, we simply use this state to render detached widgets by extracting the widget from the `canvasWidgetsStructure` in the `useDetachedChildren` hook.

PS: we are taking advantage of the fact that modal widgets can only be a child of main canvas in anvil. extracting modal states/props from `canvasWidgetsStructure` becomes more performant coz its a targetted search now.

Fixes #`Issue Number`  
_or_  
Fixes `Issue URL`
> [!WARNING]  
> _If no issue exists, please create an issue first, and check with the maintainers if the issue is valid._

## Automation

/ok-to-test tags="@tag.Anvil"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/8890517497>
> Commit: b68ad19c09f768b99d28b72626ceefaa1b79056d
> Cypress dashboard url: <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=8890517497&attempt=1" target="_blank">Click here!</a>

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **New Features**
	- Introduced a new component for rendering detached child widgets within the Anvil layout system.
- **Refactor**
	- Updated the handling and rendering process of detached widgets across various components in the Anvil system.
- **Bug Fixes**
	- Adjusted selector logic to correctly retrieve and manage detached widget states.

### CODE REVIEW

**Comment by riodeuno** [comment]
@marks0351 This PR a little hard to review. There are a few things, the rationale behind which I donot understand.

I have a few questions, but before that would you mind elaborating on what we've changed and why we've done so?

**Comment by marks0351** [comment]
> @marks0351 This PR a little hard to review. There are a few things, the rationale behind which I donot understand.
> 
> I have a few questions, but before that would you mind elaborating on what we've changed and why we've done so, in the PR description, or in a related issue?

My bad, have updated the PR description, let me know if that helps.

File: app/client/src/layoutSystems/anvil/integrations/sagas/utils.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import type { ReduxAction } from "@appsmith/constants/ReduxActionConstants";
+import { call, select } from "redux-saga/effects";
+import { getIsAnvilLayout } from "../selectors";
+
+export function* callSagaOnlyForAnvil(saga: any, action: ReduxAction<unknown>) {
+  const isAnvilLayout: boolean = yield select(getIsAnvilLayout);
+  if (isAnvilLayout) {
+    yield call(saga, action);
+  }
+}
```
[the remark this answers is not in the document]
Comment (marks0351) [reply app/client/src/layoutSystems/anvil/integrations/sagas/utils.ts:10]: takeLeading is having type issues if I try this.

**Review by riodeuno — APPROVED** [review]

**Pull request merged**